app/apps/client/externalComponent/index.js

```diff
@@ -0,0 +1,146 @@
+import { Meteor } from 'meteor/meteor';
+import { Session } from 'meteor/session';
+import { Template } from 'meteor/templating';
+
+import { APIClient } from '../../../utils';
+import { getUserAvatarURL } from '../../../utils/lib/getUserAvatarURL';
+import { setItem, getItem, getAll, clear, removeItem } from './persistence';
+
+Meteor.startup(function() {
+	window.addEventListener('message', async ({ data, source }) => {
```

Comment:
Just a comment here, we can work on that later on.

I think this part of the code should live in the Apps-Engine, as a `AppsEngineUIHost` or something similar. Then, we can keep the communication logic in one repo and make sure they are compatible for every Apps-Engine version.

As it is currently implemented, apps would have not only a dependency to the Apps-Engine version, _but a dependency to Rocket.Chat's version as well_, and things will begin to get "messier", harder to maintain.

Of course, if we move this part into the Apps-Engine, we are going to need some dependencies to be injected in the `AppsEngineUIHost` class, as it won't have easy access to `Meteor` variable, for instance. But we are going to benefit from moving away from tight coupling to the `Meteor` framework, which is a big plus :)